presto-cache/src/main/java/com/facebook/presto/cache/alluxio/AlluxioCachingFileSystem.java

```diff
@@ -64,115 +46,25 @@ public AlluxioCachingFileSystem(ExtendedFileSystem dataTier, URI uri, boolean ca
-            // Connection details in the URI has the highest priority
-            newConfiguration.merge(configurationFromUri, Source.RUNTIME);
-            mAlluxioConf = newConfiguration;
-
-            // Handle metrics
-            Properties metricsProperties = new Properties();
-            for (Map.Entry<String, String> entry : configuration) {
-                metricsProperties.setProperty(entry.getKey(), entry.getValue());
-            }
-            MetricsSystem.startSinksFromConfig(new MetricsConfig(metricsProperties));
-            mFileSystem = new LocalCacheFileSystem(new AlluxioCachingClientFileSystem(fileSystem, mAlluxioConf), mAlluxioConf);
-            super.initialize(uri, configuration);
-        }
-
-        public FSDataInputStream openFile(Path path, HiveFileContext hiveFileContext)
-                throws Exception
-        {
+            FileStatus fileStatus = dataTier.getFileStatus(path);
             // URIStatus is the mechanism to pass the hiveFileContext to the source filesystem
-            URIStatus uriStatus = mFileSystem.getStatus(getAlluxioPath(path));
-            AlluxioURIStatus alluxioURIStatus = new AlluxioURIStatus(uriStatus.getFileInfo(), hiveFileContext);
-            FileInStream fileInStream = mFileSystem.openFile(alluxioURIStatus, OpenFilePOptions.getDefaultInstance());
-            return new FSDataInputStream(new AlluxioCachingHdfsFileInputStream(
-                    fileInStream,
-                    (cacheValidationEnabled ? Optional.of(fileSystem.openFile(path, hiveFileContext)) : Optional.empty()),
-                    cacheValidationEnabled));
-        }
-
-        @Override
-        public String getScheme()
-        {
-            return uri.getScheme();
-        }
-
-        @Override
-        protected boolean isZookeeperMode()
-        {
-            return mFileSystem.getConf().getBoolean(PropertyKey.ZOOKEEPER_ENABLED);
-        }
-
-        @Override
-        protected Map<String, Object> getConfigurationFromUri(URI uri)
-        {
-            // local cache doesn't use URI authority for connection.
-            return ImmutableMap.of();
-        }
-
-        @Override
-        protected void validateFsUri(URI uri)
-        {
-        }
-
-        @Override
-        protected String getFsScheme(URI uri)
-        {
-            return getScheme();
-        }
-
-        @Override
-        protected AlluxioURI getAlluxioPath(Path path)
-        {
-            return new AlluxioURI(path.toString());
-        }
-
-        @Override
-        protected Path getFsPath(String uriHeader, URIStatus fileStatus)
-        {
-            return new Path(uriHeader + fileStatus.getPath());
+            AlluxioURIStatus alluxioURIStatus = new AlluxioURIStatus(fileStatus, hiveFileContext);
+            FSDataInputStream cachingInputStream = localCacheFileSystem.open(alluxioURIStatus, 65536 /* not used */);
```

Comment:
Just curious where this 65536 came from 🤔. Maybe we could make it 0 since it's not used anyway?

Reply:
this is a buffer size internal in the input stream. I double checked the src and it is actually used in certain corner cases. Updated

Reply:
I will remove this method with no bufferSize as arg in next version

Reply:
Since this is going to be removed soon, I'm would not be too picky on this. Ideally it should be a static constant variable or at least have some comment about the magic number :)